### PR TITLE
View refactor

### PR DESCRIPTION
**Pull request by fourtooths**

### CODE REVIEW

File: game/core/src/main/java/com/gg/game/view/AbstractView.java [hunk 1]
```diff
@@ -11,7 +12,9 @@
 
     final World world;
     SpriteBatch spriteBatch;
+    BackgroundMusic backgroundMusic;
     final Assets asset; //Used for binding objects to textures
+    private final static String Texture_Missing_MSG = "Missing texture for object";
```
Comment (Riddarvid) [inline game/core/src/main/java/com/gg/game/view/AbstractView.java]: Should be camel case

File: game/core/src/main/java/com/gg/game/model/characters/Enemy.java [hunk 2]
```diff
@@ -10,17 +10,21 @@
     public final static int Enemy_HitPoints = 20;
     private final static int Max_Speed_Value = 40;
```
Comment (Riddarvid) [inline game/core/src/main/java/com/gg/game/model/characters/Enemy.java]: Should be camel case

File: game/core/src/main/java/com/gg/game/view/Assets.java [hunk 3]
```diff
@@ -3,49 +3,64 @@
...
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 /**
  * Handles textures for objects.
  */
 class Assets {
+    private static final String Img_Map_Path = "images/";
```
Comment (Riddarvid) [inline game/core/src/main/java/com/gg/game/view/Assets.java]: Should be camel case

File: game/core/src/main/java/com/gg/game/view/BackgroundMusic.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package com.gg.game.view;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.audio.Music;
+
+
+public class BackgroundMusic {
+    private final static String Sound_Map_Path = "sounds/";
```
Comment (Riddarvid) [inline game/core/src/main/java/com/gg/game/view/BackgroundMusic.java]: Cc

File: game/core/src/main/java/com/gg/game/model/World.java [hunk 5]
```diff
@@ -23,6 +23,8 @@
     private static volatile World soleInstance; //Only instance of class. Thread safe.
     private static final int World_Width = 640;
     private static final int World_Height = 480;
+    private static final int Player_Starting_X = 150;
+    private static final int Player_Starting_Y = 150;
```
Comment (Riddarvid) [inline game/core/src/main/java/com/gg/game/model/World.java]: Cc

File: game/core/src/main/java/com/gg/game/view/PlayingStrategyView.java [hunk 6]
```diff
@@ -38,45 +40,51 @@ void loadTextures() {
...
+            int srcX = 0;
+            int srcY = 0;
+            int srcWidth = texture.getWidth();
+            int srcHeight = texture.getHeight();
+            boolean flipX = false;
+            boolean flipY = false;
+            spriteBatch.draw(texture, x, y, originX, originY, width, height, scaleX, scaleY, rotation, srcX,
+                    srcY, srcWidth, srcHeight, flipX, flipY);
+        }catch(FileNotFoundException e){
+            System.out.println(getTexture_Missing_MSG());
```
Comment (Riddarvid) [inline game/core/src/main/java/com/gg/game/view/PlayingStrategyView.java]: Cc

File: game/core/src/main/java/com/gg/game/model/World.java [hunk 7]
```diff
@@ -23,6 +23,8 @@
     private static volatile World soleInstance; //Only instance of class. Thread safe.
     private static final int World_Width = 640;
     private static final int World_Height = 480;
```
Comment (pudkipz) [inline game/core/src/main/java/com/gg/game/model/World.java]: ~~World width and height should be taken from constants class.

We could also consider adding the starting position to the constants class.~~

Oops, I thought we'd already added the magic numbers class.

File: game/core/src/main/java/com/gg/game/view/AbstractView.java [hunk 8]
```diff
@@ -34,17 +38,54 @@ public void create() {
...
+        loadObjectTextures();
+    }
+
+    abstract void loadStaticTextures();
+
+    abstract void loadObjectTextures();
+
+    abstract void setBackgroundMusic();
+
+    String getTexture_Missing_MSG(){
```
Comment (pudkipz) [inline game/core/src/main/java/com/gg/game/view/AbstractView.java]: Mixing camel and snake case.

Suggested name: getTextureMissingMessage alt. getMissingTextureMessage.

**Pull request merged**